hypothesis-python/tests/cover/test_annotations.py

```diff
@@ -124,6 +125,8 @@ def test_attrs_inference_builds(c):
     pass
 
 
-@given(st.from_type(Inferrables))
-def test_attrs_inference_from_type(c):
-    pass
+with pytest.warns(SmallSearchSpaceWarning):
+
+    @given(st.from_type(Inferrables))
+    def test_attrs_inference_from_type(c):
+        pass
```

Comment:
Hmm, we'd prefer to trigger this warning when the test function is _executed_ rather than when it's _defined, as for other strategy errors.  Might be handleable via using a warnings filter with "error" around the first "eager" call, but not when wrapped in `LazyStrategy`?

We'd then write this test

```python
def test_attrs_inference_from_type(c):
    s = st.from_type(Inferrables)
    with pytest.warns(SmallSearchSpaceWarning):
        s.example()
```

Reply:
Sure, will do. And in this particular case I guess the @given` to `.example()` is fine since there is no assert.

Reply:
...actually... warnings filters are not thread safe, should we try to be?

Reply:
Nah, our test suite doesn't need to be threadsafe.

Reply:
Yep, but the simplest solution (deferring warnings by executing under warnings filter, and convert to Lazy on exception - which is already done) will render other peoples' test suites unsafe as well if they use `from_type`. Even if they don't receive warnings from `from_type`, but other warnings in other threads.

A thread safe version will maybe have to add a new argument whether to raise or warn. Which is not catastrophic, but uglier and also less generic.

Reply:
...but then again, if they deal with other warnings in other threads their test suites are likely already thread-unsafe. So the major concern is if they do use from_type.

Reply:
Ouch, yeah, I misunderstood before (early morning!).

I think that "if your tests need to handle warnings they are not thread safe" is the way to go though, and we can try something fancier if people complain.